backend/src/ee/services/secret-approval-policy/secret-approval-policy-service.ts

```diff
@@ -301,7 +305,14 @@ export const secretApprovalPolicyServiceFactory = ({
       });
     }
 
-    await secretApprovalPolicyDAL.deleteById(secretPolicyId);
+    await secretApprovalPolicyDAL.transaction(async (tx) => {
+      await secretApprovalRequestDAL.update(
+        { policyId: secretPolicyId, status: RequestState.Open },
+        { status: RequestState.Closed },
+        tx
+      );
+      await secretApprovalPolicyDAL.softDeleteById(secretPolicyId, tx);
+    });
```

Comment:
Nit: Currently we are doing `return sapPolicy`, which was okay before because `sapPolicy` wouldn't change. Now we are actually updating the policy, so we should return the updated policy instead.

You can do `return await secretApprovalPolicyDAL.softDeleteById(secretPolicyId, tx);` inside the transaction. And do `updatedPolicy = await secretApprovalPolicyDAL.transaction(async (tx) => { .....`